fog/overseer/server/src/worker.rs

```diff
@@ -496,30 +515,26 @@ where
         let result = retry_with_index(
             Fixed::from_millis(200).take(Self::NUMBER_OF_TRIES),
             |current_try| {
-                match self.ingest_clients[activated_node_index].activate() {
+                let ingest_client = &self.ingest_clients[activated_node_index];
+                match ingest_client.activate() {
                     Ok(_) => {
                         log::info!(
                             self.logger,
-                            "Node at index {} successfully activated.",
-                            activated_node_index
+                            "Node {} successfully activated.",
+                            ingest_client.get_uri(),
                         );
                         OperationResult::Ok(())
                     }
                     // TODO: Alert Ops to take manual action at this point.
                     Err(err) => {
                         let number_of_remaining_tries =
                             Self::NUMBER_OF_TRIES - current_try as usize;
-                        let error_message = match number_of_remaining_tries {
-                            0 => format!(
-                                "Did not succeed in setting a new key on node at index {}. Underlying error: {}",
-                                activated_node_index,
-                                err
-                            ),
-                            _ => format!(
-                                "Node at index {} not activated. Will try {} more times. Underlying error: {}",
-                                activated_node_index, number_of_remaining_tries, err
-                            ),
-                        };
+                        let error_message = format!(
+                            "Node {} not activated. Will try {} more times. Underlying error: {}",
+                            ingest_client.get_uri(),
+                            number_of_remaining_tries,
+                            err
+                        );
                         OperationResult::Retry(OverseerError::ActivateNode(error_message))
```

Comment:
... On further thought, do we still want to return Retry when num_retries = 0? If not, then we might as well change back to using a `match`, and returning a different message and result when num_retries=0.

Reply:
That was the original reason why I kept the messages as they were, but I think it is fine as is. And IIUC, I think that it does not matter if we return a Retry when num_tries is 0 because the logic on line 497 will ensure that it is not retried after the num_tries is 0.